// crud_service/user.proto

syntax = "proto3";

package user;

service UserService {
    rpc GetUser (UserRequest) returns (UserResponse);
    rpc GetUsers (EmptyRequest) returns (UserListResponse);
    rpc UpdateUser (UpdateUserRequest) returns (UserResponse);
    rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse);
    rpc RegisterUser (RegisterUserRequest) returns (UserResponse);
     rpc ValidateUser (LoginRequest) returns (LoginResponse);
}

message UserRequest {
    int32 id = 1;
}

message RegisterUserRequest {
    string username = 1;
    string first_name = 2;
    string last_name = 3;
    string password = 4;
}

message UpdateUserRequest {
    int32 id = 1;
    string username = 2;
    string first_name = 3;
    string last_name = 4;
    string password = 5;
}

message DeleteUserRequest {
    int32 id = 1;
}

message DeleteUserResponse {
    bool success = 1;
}

message UserResponse {
    int32 id = 1;
    string username = 2;
    string first_name = 3;
    string last_name = 4;
}

message UserListResponse {
    repeated UserResponse users = 1;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}
message LoginResponse {
    bool valid = 1;
    string message = 2;
}
message EmptyRequest {}
